proof/refine/ARM/KHeap_R.thy

```diff
@@ -3910,4 +3910,41 @@ lemma threadGet_getObject:
   apply (simp add: threadGet_def threadRead_def oliftM_def getObject_def[symmetric])
   done
 
+lemma isBlockedOnSend_def2:
+  "isBlockedOnSend st = (\<exists>v0 v1 v2 v3 v4. st = Structures_H.thread_state.BlockedOnSend v0 v1 v2 v3 v4)"
```

Comment:
I believe that `isBlockedOnSend` and `isSend` both exist in the Haskell and are needed for it to be executable. Though this does reveal some duplication which could be removed. Since Haskell doesn't know about datatype packages, I don't think we can remove it from `is_BlockedOnSend` Haskell entirely.